lib/mongoid/extensions/binary.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Mongoid
+  module Extensions
+    module Binary
```

Comment:
Could this module be `BSONBinary` or similar for clarity?

Reply:
Or maybe actually nest it in `BSON` submodule to get `Mongoid::Extensions::BSON::Binary`.

Reply:
done

Reply:
Just a note, now when BSON is used in /extensions module it needs to be prefixed with ::